lib/listen/record.rb

```diff
@@ -75,31 +63,36 @@ def build
 
     private
 
+    # TODO: refactor/refactor out
+    def add_dir(dir, rel_path, children)
+      rel_path = '.' if [nil, '', '.'].include? rel_path
+      dirname, basename = Pathname(rel_path).split.map(&:to_s)
+      basename = '.' if [nil, '', '.'].include? basename
+      if [nil, '', '.'].include?(dirname) && basename == '.'
+        @paths[dir.to_s]['.'] = Hash[children.map { |child| [child, {} ] } ]
+      else
+        @paths[dir.to_s][rel_path] ||= {}
```

Comment:
`_fast_build` should set things up so that '.' actually contains entries when using `dir_entries` - that's what `add_dir` is for here. This is for the `previous` to return something in `Directory.scan` on the first call.

> Also might be a mistake here either way as it only does the children for the root folder?

That's why I marked it with TODOs - it should be part of the initial build only.

> Lastly the bug with the moving of a folder not showing the old folder in the removed list is still present 

This needs a separate acceptance test.